lib/ConversionProvider.php

```diff
@@ -0,0 +1,141 @@
+<?php
+/**
+ * SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace OCA\Pandoc;
+
+use OCP\AppFramework\Services\IAppConfig;
+use OCP\Files\Conversion\ConversionMimeProvider;
+use OCP\Files\Conversion\IConversionProvider;
+use OCP\Files\File;
+use OCP\IConfig;
+use OCP\IL10N;
+use RuntimeException;
+
+class ConversionProvider implements IConversionProvider {
+	private ?array $types = null;
+
+	public function __construct(
+		private IL10N $l10n,
+		private IConfig $config,
+		private IAppConfig $appConfig,
+	) {
+	}
+
+	private function getPDFEngine(): string {
+		if (method_exists($this->appConfig, 'getValueString')) {
+			return $this->appConfig->getValueString('pdf_engine', '');
```

Comment:
I don't know how many compatible PDF engines exist out there, but if pdflatex is a very common one, we could also try to detect if it's installed and default to it in that case to make the setup one step less? :thinking:

Reply:
Done now with the binary detector we have in the OCP API.